File: core/src/com/oni/donjon/generator/ImproveRectangle.java
package com.oni.donjon.generator;

/**
 * @author Daniel Chesters (on 13/02/16).
 */
public class ImproveRectangle extends com.badlogic.gdx.math.Rectangle {

    public ImproveRectangle(float x, float y, float w, float h) {
        super(x, y, w, h);
    }

    public boolean intersects(ImproveRectangle r) {
        return x <= r.x + r.width && x + width >= r.x && y <= r.y + r.height && y + height >= r.y;
    }
}

File: core/src/main/kotlin/com/oni/donjon/generator/ImproveRectangle.kt
package com.oni.donjon.generator

import com.badlogic.gdx.math.Rectangle

/**
 * @param r the other {@link Rectangle}
 * @return whether this rectangle overlaps the other rectangle or the two rectangle are side by side.
 * @author Daniel Chesters (on 30/05/2017).
 *
 */
fun Rectangle.intersects(r: Rectangle): Boolean =
        x <= r.x + r.width && x + width >= r.x && y <= r.y + r.height && y + height >= r.y
